src/components/smart/Table.vue

```diff
@@ -1,27 +1,79 @@
 <template>
-  <div class="overflow-auto border shadow-md rounded-md border-dividerDark">
-    <table class="w-full">
-      <thead>
-        <slot name="head">
-          <tr
-            class="text-sm text-left border-b border-dividerDark bg-primaryLight text-secondary"
-          >
+  <div v-if="pagination" class="mb-3 flex justify-end">
+    <div class="flex w-min">
+      <HoppButtonSecondary
+        outline
+        filled
+        :icon="IconLeft"
+        :disabled="page === 1"
+        @click="changePage(PageDirection.Previous)"
+      />
+
+      <div class="flex h-full w-10 items-center justify-center">
+        <p>{{ page }}</p>
+      </div>
+
+      <HoppButtonSecondary
+        outline
+        filled
+        :icon="IconRight"
+        :disabled="page === pagination.totalPages"
+        @click="changePage(PageDirection.Next)"
+      />
+    </div>
+  </div>
+
+  <div class="overflow-auto rounded-md border border-dividerDark shadow-md">
+    <div v-if="searchBar" class="flex w-full items-center bg-primary">
+      <icon-lucide-search class="ml-3 text-xs" />
+      <input
+        v-model="searchQuery"
+        class="h-full w-full bg-primary p-3"
+        :placeholder="searchBar.placeholder ?? 'Search...'"
+      />
+    </div>
+
+    <div v-if="isSpinnerEnabled" class="mx-auto my-3 h-5 w-5 text-center">
+      <HoppSmartSpinner />
+    </div>
+
+    <table v-else-if="list" class="w-full">
+      <thead v-if="list.length > 0">
+        <tr
+          class="border-b border-dividerDark bg-primaryLight text-left text-sm text-secondary"
+        >
+          <th v-if="checkbox" class="w-5 pl-6 pt-1">
```

Comment:
```suggestion
          <th v-if="checkbox" class="flex justify-center items-center py-2 my-1 flex-1" >
```

Reply:
@nivedin When flex is applied here, the height reduces from filling the entire row height to just the minimum required height. However, that results in the checkbox being centered only within that small height and looks out of place. I think just a horizontal padding here will do the trick since the `td` automatically assumes the height of the rows.

Reply:
@JoelJacobStephen  can you check if this works

` <th v-if="checkbox" class="py-2">
              <div class="flex items-center justify-center h-full">
                <input
                  ref="selectAllCheckbox"
                  type="checkbox"
                  :checked="areAllRowsSelected"
                  class="h-full w-full"
                  @click.stop="toggleAllRows"
                />
              </div>
            </th>`

Reply:
@nivedin This works. However, there is no horizontal width. So I will just add a small horizontal padding.